util/dynamicConfig.js

```diff
@@ -1,29 +1,62 @@
+/* eslint-disable no-param-reassign */
 const get = require("get-value");
 const unset = require("unset-value");
 
-function getDynamicConfig(event) {
-  const { Config } = event.destination;
-  if (Config) {
-    Object.keys(Config).forEach(field => {
-      let value = Config[field].toString().trim();
-      if (value.startsWith("{{") && value.endsWith("}}")) {
-        value = value.replace("{{", "");
-        value = value.replace("}}", "");
-        if (value.includes("||")) {
-          const path = value.split("||")[0].trim();
-          const getFieldVal = get(event, path);
-          if (getFieldVal) {
-            Config[field] = getFieldVal;
-            unset(event, path);
-          } else {
-            Config[field] = JSON.parse(value.split("||")[1].trim()).toString();
-          }
-        }
+function isValidPattern(val) {
+  // this rgegex checks for pattern  "only spaces {{ path || defaultvalue }}  only spaces" .
+  //  " {{message.traits.key  ||   \"email\" }} "
+  //  " {{ message.traits.key || 1233 }} "
+  const re = /^\s*\{\{\s*(?<path>[a-zA-Z_]([a-zA-Z0-9_]*\.[a-zA-Z_][a-zA-Z0-9_]*)+)+\s*\|\|\s*"?(?<defaultValue>.*)"?\s*\}\}\s*$/;
+  return re.test(String(val));
+}
+
+function getDynamicConfigValue(event, value) {
+  value = value.replace("{{", "").replace("}}", "");
+  if (value.includes("||")) {
+    const path = value.split("||")[0].trim();
+    const fieldVal = get(event, path);
+    if (fieldVal) {
+      value = fieldVal;
+      unset(event, path);
+    } else {
+      value = JSON.parse(value.split("||")[1].trim());
```

Comment:
We removed .toString() since we don’t want to restrict the value to strings only. So that defaultValue can be numbers/boolean as well. @ruchiramoitra @saikumarrs

Reply:
@abandonned I don't think we need JSON parsing here.

Reply:
yeah we can remove JSON.parse. But should we keep `toString()` as suggested by @ruchiramoitra di?

Reply:
Yes

Reply:
We should provide this information in doc that even though we accept all types of value as defaultVal, we are converting to string while configuring.